TaskManagementSystem/main.go

```diff
@@ -7,27 +7,58 @@ import (
 	"time"
 )
 
-func session2() {
-	var ProjectStatus string = "in progress"
+func TaskUpdate(CompletedTasks int) {
+
 	const NumberOfTotalTasks int = 100
-	CompletedTasks := 25
-	const (
-		Low = iota
-		Medium
-		High
-	)
-	var ProjectCompletionStatus bool = true
+	var ProjectStatus string
+
+	RemainingTasks := NumberOfTotalTasks - CompletedTasks
+	if CompletedTasks != NumberOfTotalTasks {
+		ProjectStatus = "in progress"
+	} else {
+		ProjectStatus = "finished"
+	}
+
+	var ProjectCompletionStatus bool
 	if ProjectStatus == "in progress" {
 		ProjectCompletionStatus = false
 	} else {
 		ProjectCompletionStatus = true
 	}
-	fmt.Println("Current project status:", strings.ToUpper(ProjectStatus))
-	fmt.Printf("Tasks completed: %d out of %d\n", CompletedTasks, NumberOfTotalTasks)
+
+	const (
+		Low = iota + 1
+		Medium
+		High
+	)
+
+	var ProjectPhase string
+	switch {
+	case CompletedTasks < 30:
+		ProjectPhase = "Starting phase"
+	case 30 <= CompletedTasks && CompletedTasks <= 60:
+		ProjectPhase = "Midway"
+	case 60 <= CompletedTasks:
+		ProjectPhase = "Final phase"
+	default:
+		fmt.Println("Wrong entered data")
+	}
```

Comment:
There is a logical and syntax issues here:

1. In syntax, first set variable and then comparison (ex: CompletedTasks >=60, neighter 60<= CompletedTask). This will set readability of the code
2. less than 60 (included) is Midway, so more than 60 (not included) is Final. So no need to also set double case as equal
```suggestion
	switch {
	case CompletedTasks < 30:
		ProjectPhase = "Starting phase"
	case CompletedTasks >= 30 && CompletedTasks <= 60:
		ProjectPhase = "Midway"
	case CompletedTasks > 60:
		ProjectPhase = "Final phase"
	default:
		fmt.Println("Wrong entered data")
	}
```